provider/filestore.go

```diff
@@ -244,9 +244,9 @@ func NewS3FileStore(config Config) (FileStore, error) {
 		return nil, fmt.Errorf("could not deserialize s3 store config: %v", err)
 	}
 
-	s3StoreConfig.BucketPath = strings.TrimPrefix("s3://", strings.TrimPrefix(s3StoreConfig.BucketPath, "s3a://"))
+	trimmedBucket := strings.TrimPrefix(strings.TrimPrefix(s3StoreConfig.BucketPath, "s3a://"), "s3://")
```

Comment:
ooooof that's a bad error lol. We should have had a test for this

Reply:
also I would've maybe split this into two lines, it took me a sec to figure out what was going on

Reply:
fair

Reply:
This isn't really where I want it, but theres 2 other tasks to fix this for other filestores where we can put it in a better place